app/lib/frontend/email_sender.dart

```diff
@@ -168,6 +174,30 @@ class _GmailSmtpRelay implements EmailSender {
     }
   }
 
+  Future<_GmailConnection> _getConnection() async {
+    final old = _connection == null ? null : await _connection;
+    if (old == null || old.isExpired) {
+      _connection = Future.microtask(
+        () async {
+          if (old != null) {
+            try {
+              await old.connection.close();
+            } catch (e, st) {
+              _logger.warning('Unable to close SMTP connection.', e, st);
+            }
+          }
+          return _GmailConnection(
+            PersistentConnection(
+              await _getSmtpServer(),
+              timeout: Duration(seconds: 15),
+            ),
+          );
+        },
+      );
+    }
+    return _connection!;
+  }
```

Comment:
Always return early, and do `return _connection = ` then you won't need null assertions :D
```suggestion
  Future<_GmailConnection> _getConnection() async {
    final old = _connection == null ? null : await _connection;
    if (old != null && !old.isExpired) {
      return old;
    }
    return _connection = Future.microtask(() async {
      if (old != null) {
        try {
          await old.connection.close();
        } catch (e, st) {
          _logger.warning('Unable to close SMTP connection.', e, st);
        }
      }
      return _GmailConnection(
        PersistentConnection(
          await _getSmtpServer(),
          timeout: Duration(seconds: 15),
        ),
      );
    });
  }
```